framework/src/main/java/org/apache/drill/test/framework/Utils.java

```diff
@@ -167,9 +168,25 @@
           }
         }
         if (!foundTests) {continue;}
-        
-        String queryFileExtension = modeler.matrices.get(0).inputFile;
+       
+	String queryFileExtension = modeler.matrices.get(0).inputFile;
         String expectedFileExtension = modeler.matrices.get(0).expectedFile;
+
```

Comment:
changed the regex string to include .fail/.failed or any other failextension provided in the testDefFile